Custom error documentation (#10937)
ethereum/solidity: docs/examples/blind-auction.rst

        event HighestBidIncreased(address bidder, uint amount);
        event AuctionEnded(address winner, uint amount);

        // Errors that describe failures.

        // The triple-slash comments are so-called natspec
        // comments. They will be shown when the user
        // is asked to confirm a transaction or
        // when an error is displayed.

        /// The action has already ended.
        error AuctionAlreadyEnded();

>         /// There is already a higher bid.

Reviewer: ```suggestion
        /// There is already a higher or equal bid.
```

>         /// The action has already ended.
>         error AuctionAlreadyEnded();
>         /// There is already a higher bid.
>         error BidNotHighEnough(uint highestBid);
>         /// The action has not ended yet.
>         error AuctionNotYetEnded();
>         /// The function actionEnd has already been called.
>         error AuctionEndAlreadyCalled();

Reviewer: ```suggestion
        /// The auction has already ended.
        error AuctionAlreadyEnded();
        /// There is already a higher bid.
        error BidNotHighEnough(uint highestBid);
        /// The auction has not ended yet.
        error AuctionNotYetEnded();
        /// The function auctionEnd has already been called.
        error AuctionEndAlreadyCalled();
```

Reviewer: I see this is fixed now except for `actionEnd` (which I think should be `auctionEnd()`.

            // Revert the call if the bidding
            // period is over.
            if (block.timestamp > auctionEndTime)
                revert AuctionAlreadyEnded();

            // If the bid is not higher, send the
            // money back (the revert statement
            // will revert all changes in this
            // function execution including
            // it having received the money).

>             if (msg.value >= highestBid)

Reviewer 2: `<=`?

            // external contracts.

            // 1. Conditions

>             if (block.timestamp > auctionEndTime)

Reviewer 2: `<`

        event AuctionEnded(address winner, uint highestBid);

        // Errors that describe failures.

        /// The function has been called too early.
        /// Try again at `time`.
        error TooEarly(uint time);
        /// The function has been called too late.
        /// It cannot be called after `time`.
        error TooLate(uint time);

>         /// The function actionEnd has already been called.

Reviewer: ```suggestion
        /// The function auctionEnd has already been called.
```

        event AuctionEnded(address winner, uint highestBid);

        // Errors that describe failures.

        /// The function has been called too early.

>         /// Try again at `_time`.

Reviewer 3: ```suggestion
        /// Try again at @param _time.
```

Reviewer 3: and similarly below

Author: This is not the way to use this tag.

>         error TooEarly(uint _time);
>         /// The function has been called too late.
>         /// It cannot be called after `_time`.
>         error TooLate(uint _time);

Reviewer: Other errors you defined in other places do not have the `_` prefix:
```suggestion
        error TooEarly(uint time);
        /// The function has been called too late.
        /// It cannot be called after `_time`.
        error TooLate(uint time);
```

This reminds me: you should update the Solidity coding style page to mention errors and the recommended naming convention (including parameter names).
